common/azure-common/src/main/java/com/azure/common/implementation/SwaggerMethodParser.java

```diff
@@ -149,21 +140,18 @@ else if (TypeUtil.isTypeOrSubTypeOf(returnValueWireType, List.class)) {
             expectedStatusCodes = expectedResponses.value();
         }
 
-        final UnexpectedResponseExceptionType unexpectedResponseExceptionType = swaggerMethod.getAnnotation(UnexpectedResponseExceptionType.class);
-        if (unexpectedResponseExceptionType == null) {
-            exceptionType = ServiceRequestException.class;
-        }
-        else {
-            exceptionType = unexpectedResponseExceptionType.value();
+        for (UnexpectedResponseExceptionType exceptionAnnotation : swaggerMethod.getAnnotationsByType(UnexpectedResponseExceptionType.class)) {
+            UnexpectedException exception = new UnexpectedException(exceptionAnnotation.value());
+            if (exceptionAnnotation.code().length == 0) {
+                exceptionMapping.putIfAbsent(FALLTHROUGH_EXCEPTION_CODE, exception);
+            } else {
+                for (int statusCode : exceptionAnnotation.code()) {
+                    exceptionMapping.putIfAbsent(statusCode, exception);
+                }
+            }
```

Comment:
Is it necessary to do this work? Could it be delayed until it is needed (i.e when an exception is needed to be thrown)?

Reply:
Construction of the mapping is now done when the first exception response is seen.